syntax = "proto3";

package gaps;

message TofEvent {
    float  gcutime               = 1;
    uint64 eventid               = 2;
    float  start_time            = 3;
    float  time_of_flight        = 4;
    float  beta                  = 5;
    uint32 nhits                 = 6; // extra nhits field for db queries
    //all of these have one entry per paddle
    repeated uint64 clk_cycl     = 7 [packed = true];
    repeated TofHit hits         = 8;
    bool good                    = 9;
    string run_id                = 10;
}

message TofHit {
   uint32 event_id        = 1;
   uint32 channelA_no     = 2;
   float  chargeA         = 3; 
   float  chargeB         = 4;
   float  peak_timeA      = 5;
   float  peak_timeB      = 6;
   float  peak_heightA    = 7;
   float  peak_heightB    = 8;
   float  t_at_thrA       = 9;
   float  t_at_thrB       = 10;
   float  d_longitude     = 12;
   float  t0              = 13;
   // FIXME - technically, this event is not reduced anymore
   repeated float wfA     = 14   [packed = true];
   repeated float wfB     = 15   [packed = true];
   repeated float timesA  = 16   [packed = true];
   repeated float timesB  = 17   [packed = true];
   string paddle_id       = 18;
   float  t0_thr          = 19;
   float  t0_cfd          = 20;
   float  t_at_cfdA       = 21;
   float  t_at_cfdB       = 22;
   float  ballooo         = 24;
   // defined by UCLA 1 = A, 2 = B, 3 = BOTH
   uint32 hit_mask        = 23;
}

